src/storage/txn/commands/check_txn_status.rs

```diff
@@ -38,3 +46,46 @@ impl CommandExt for CheckTxnStatus {
     write_bytes!(primary_key);
     gen_lock!(primary_key);
 }
+
+impl<S: Snapshot, L: LockManager, P: PdClient + 'static> WriteCommand<S, L, P> for CheckTxnStatus {
+    fn process_write(
+        &mut self,
+        snapshot: S,
+        lock_mgr: &L,
+        pd_client: Arc<P>,
+        _extra_op: ExtraOp,
+        statistics: &mut Statistics,
+        _pipelined_pessimistic_lock: bool,
+    ) -> Result<WriteResult> {
+        let mut txn = MvccTxn::new(
+            snapshot,
+            self.lock_ts,
+            !self.ctx.get_not_fill_cache(),
+            pd_client,
+        );
+
+        let mut released_locks = ReleasedLocks::new(self.lock_ts, TimeStamp::zero());
+        let (txn_status, released) = txn.check_txn_status(
+            self.primary_key.clone(),
```

Comment:
@nrc Here's an example of the clone
Another example is `self.ctx.clone` that used everywhere in read and write processes.